### ISSUE TITLE
 buffer exceeded errors

### ISSUE THREAD

**Issue by samjustus** [rancher/rancher#36584]
Breakout issue of rancher/rancher#34819

Rancher Cluster:
Rancher version:2.5.12-rc3
Number of nodes:1500
Node OS version: RancherOS- v1.5.6
RKE/RKE2/K3S version: RKE
Kubernetes version: 1.18.6

Downstream Cluster:
Number of Downstream clusters: 150
Node OS: RancherOS-v1.5.6
RKE/RKE2/K3S version: RKE
Kubernetes version:v1.18.20
CNI:

Other:
Underlying Infrastructure: Azure
Any 3rd party software installed on the nodes: NA
Customer’s main time zone: UTC +2
   
**Describe the bug**
While running the script I see free memory on the DO host decline again, but it doesn't appear to be detrimental. Typically always at least 400MB free/cached. Likely not an issue.

Rancher server logs snippet below. Notice this is similar to my snippet prior when I reproduced. This is probably because the snippet prior wasn't related to the actions being performed by the script. We now see remotedialer buffer exceeded errors. There are a ton of these errors, this is just a snippet of them.

```
2022/01/14 18:03:47 [ERROR] remotedialer buffer exceeded, length: 4215507
2022/01/14 18:03:47 [ERROR] remotedialer buffer exceeded, length: 4248275
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4206503
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4239271
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4272039
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4304807
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4337575
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4370343
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4403111
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4435879
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4468647
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4501415
2022/01/14 18:03:56 [ERROR] remotedialer buffer exceeded, length: 4509027
```

**Comment by jbilliau-rcd** [comment]
I just upgraded from 2.5.11 to 2.5.12 and am seeing this error spammed in the logs quite often. Is this a problem? Everything "seems" to be fine so far.....though it's only been 10 min or so.

**Comment by wzrdtales** [comment]
this is somehow related to the charts it seems

**Comment by vj6** [comment]
also facing this on 


Component | Version
-- | --
Rancher | v2.5.12
User Interface | v2.5.12
Helm | v2.16.8-rancher1
Machine | v0.15.0-rancher55

Component	Version
[Rancher](https://github.com/rancher/rancher)	v2.5.12
[User Interface](https://github.com/rancher/ui)	v2.5.12
[Helm](https://github.com/rancher/helm)	v2.16.8-rancher1
[Machine](https://github.com/rancher/machine)	v0.15.0-rancher55

**Comment by jbilliau-rcd** [comment]
We upgraded to 2.6.4, still happening.

**Comment by rmweir** [comment]
Synopsis of issue: A new log was introduced under error level when it should be debug level. This is alarming customers as they are seeing a flood of errors. When the conditions occur that result in the errors, user intervention is not necessary as there is a mechanism to pause filling the buffer until it is not full.

Resolved (or not): Resolved. This error may have been observed while other issues occurred but those should be documented in separate issues.

Versions: 2.5.14. Fix will be in 2.6 too but that will be a separate issue.

**Comment by jbilliau-rcd** [comment]
@rmweir So I understand that this "error" is only supposed to be under debug level logging, but is it supposed to happen at all, especially frequently? We are seeing it spammed A LOT in our logs, hundreds of times a minute. Is it normal that a buffer fills that quickly?

I ask because our nonprod Rancher environment is becoming increasingly unstable, with Rancher pods dying at frequent intervals due to node memory pressure, which is due to the Rancher pods using ever-increasing amounts of RAM. Are these related?

**Comment by papanito** [comment]
I see the issue is closed, so in which release is this fixed?

**Comment by moio** [comment]
Fix is in [remotedialer](https://github.com/rancher/remotedialer), which is a dependency in Rancher, and it was merged on May 19 2022 as you can see here:

https://github.com/rancher/remotedialer/pull/45

You can check if a certain version of Rancher has a certain version of remotedialer by looking at the go.mod file, for example for 2.5.17:

https://github.com/rancher/rancher/blob/v2.5.17/go.mod#L104

As you see in the version number, `20220624`, this was taken on June 24, so it contains the fix. You can play with the URL above on different versions, eg. 2.5.13 does not have it:

https://github.com/rancher/rancher/blob/v2.5.13/go.mod#L102

as `20220318` is March 18, so earlier than the fix got merged.


Anyway, I see it fixed in all latest released versions: [2.5.17](https://github.com/rancher/rancher/blob/v2.5.17/go.mod#L104), [2.6.10](https://github.com/rancher/rancher/blob/v2.6.10/go.mod#L120) and [2.7.1](https://github.com/rancher/rancher/blob/v2.7.1/go.mod#L118).

As always I can recommend to update to the latest and let us know if you still experience problems!

HTH
